cli/command/container/cp.go

```diff
@@ -171,12 +237,30 @@ func copyFromContainer(ctx context.Context, dockerCli command.Cli, copyConfig cp
 		RebaseName: rebaseName,
 	}
 
+	var copySize int64
+	copySize, content = getSizeReadCloser(content)
+	content = &copyReader{ReadCloser: content, toContainer: false, containerName: copyConfig.container, copyPath: dstPath, quiet: copyConfig.quiet, copySize: copySize}
+
 	preArchive := content
+
 	if len(srcInfo.RebaseName) != 0 {
 		_, srcBase := archive.SplitPathDirEntry(srcInfo.Path)
 		preArchive = archive.RebaseArchiveEntries(content, srcBase, srcInfo.RebaseName)
 	}
-	return archive.CopyTo(preArchive, srcInfo, dstPath)
+
+	if !copyConfig.quiet {
+		fmt.Println("Preparing to copy...")
+		fmt.Print("\033[s")
+	}
+
+	res := archive.CopyTo(preArchive, srcInfo, dstPath)
+
+	if !copyConfig.quiet {
+		fmt.Println()
```

Comment:
Same as the progress output, we should have an output stream that we write to rather than assuming `os.Stdout`

Reply:
Right.